### DISCUSSION TITLE
What is the recommended way to programmatically stand up a staging Supabase DB from a production snapshot?

### DISCUSSION THREAD

**Discussion by aresnick** [supabase/supabase#4700] [Questions] [answered]
I have a production Supabase DB backing an instance of [Strapi](https://strapi.io/), and I'd like to be able to programmatically mirror this DB purposes of staging model/schema changes.  I can successfully `pg_dumpall` the production DB, but I can't restore from this because AFAIK there is no way to execute SQL as a superuser programmatically (though it could be done interactively through the dashboard).

My investigations led me to the CLI, but unfortunately I can't seem to get the CLI working either— I run into [this issue](https://github.com/supabase/cli/issues/117).

What is the recommended way to programmatically stand up a staging Supabase DB from a production snapshot?

Previously:

* ["best practice for local, development, staging tables?"](https://github.com/supabase/supabase/discussions/2590)
* ["Local development flow recommendations"](https://github.com/supabase/supabase/discussions/3277)
* ["How can we duplicate project DB?"](https://github.com/supabase/supabase/discussions/2217)

Thank you!

**Comment by bohan0** [comment]
You can temporarily set the 'postgres' user to superuser mode with `ALTER ROLE postgres SUPERUSER` (or you can create another role/user with SUPERUSER privilege). See [Migrating between projects](https://supabase.com/docs/guides/database#migrating-between-projects)

**Comment by aresnick** [comment]
@bohan0 Thank you!  But, my understanding is that to either run `ALTER ROLE postgres SUPERUSER` or create a role with `SUPERUSER` privileges, I must do this via the dashboard.  Is that right?

**Comment by bohan0** [comment]
Yes, thats right.

**Comment by soedirgo** [comment, marked as answer]
Yeah this isn't possible right now, not without interacting with the dashboard.

We're currently working on an Admin API, for creating orgs/projects for starters. Duplicating projects might be further off though.

**Reply by aresnick** [reply]
Got it; thank you!  Does that mean the recommended flow is to maintain a staging DB "by hand" and just wipe that when restoring a snapshot?

**Reply by soedirgo** [reply]
Yep, that's right. Also, I recommend sticking closely to the steps [here](https://supabase.com/docs/guides/database#migrating-between-projects) as there might be important caveats. e.g. Storage objects aren't migrated as it's outside the db, so the entries would 404 if restoring from `pg_dumpall`.
